waddrmgr/db.go

```diff
@@ -69,6 +70,24 @@ const (
 	adtScript addressType = 2
 )
 
+// falseByte and trueByte are consts used to serialize boolean values.
+const (
+	falseByte byte = iota
+	trueByte
+)
+
+// byteAsBool converts a byte to a bool.
+func byteAsBool(b byte) (bool, error) {
+	switch b {
+	case falseByte:
+		return false, nil
+	case trueByte:
+		return true, nil
+	default:
+		return false, errors.New("invalid byte representation of bool")
+	}
+}
```

Comment:
I'd be ok with simplifying this to just returning `b != 0` and removing the error.  This was more of a concern of single bit flips when writing to a flat file, but the wallet db is checksummed.